packages/emitter-framework/src/python/components/class-declaration/class-declaration.test.tsx

```diff
@@ -0,0 +1,1334 @@
+        <ClassDeclaration type={WidgetOperations} methodType="method">
+          <hbr />
+          <hbr />
+          <List>
+            <Method name="do_work" returnType="None" doc="This is a test" />
+          </List>
+        </ClassDeclaration>,
+      ]),
+    ).toRenderTo(`
+      from dataclasses import dataclass
+
+      @dataclass(kw_only=True)
+      class WidgetOperations:
+        id: str
+        weight: int
+
+        def do_work(self) -> None:
+          """
+          This is a test
+          """
+          pass
+
+
+      `);
+  });
+
+  it("creates a class with a classmethod if a model is provided, a class method is provided and methodType is set to class", async () => {
+    const { program, WidgetOperations } = await Tester.compile(t.code`
+    model ${t.model("WidgetOperations")} {
+      id: string;
+      weight: int32;
+    }
+    `);
+
+    expect(
+      getOutput(program, [
+        <ClassDeclaration type={WidgetOperations} methodType="class">
+          <hbr />
+          <hbr />
+          <List>
+            <Method name="do_work" returnType="None" doc="This is a test" />
+          </List>
+        </ClassDeclaration>,
+      ]),
+    ).toRenderTo(`
+      from dataclasses import dataclass
+
+      @dataclass(kw_only=True)
+      class WidgetOperations:
+        id: str
+        weight: int
+
+        @classmethod
+        def do_work(cls) -> None:
+          """
+          This is a test
+          """
+          pass
+
+
+      `);
+  });
+
+  it("creates a class with a staticmethod if a model is provided, a class method is provided and methodType is set to static", async () => {
+    const { program, WidgetOperations } = await Tester.compile(t.code`
+    model ${t.model("WidgetOperations")} {
+      id: string;
+      weight: int32;
+    }
+    `);
+
+    expect(
+      getOutput(program, [
+        <ClassDeclaration type={WidgetOperations} methodType="static">
+          <hbr />
+          <hbr />
+          <List>
+            <Method name="do_work" returnType="None" doc="This is a test" />
+          </List>
+        </ClassDeclaration>,
+      ]),
+    ).toRenderTo(`
+      from dataclasses import dataclass
+
+      @dataclass(kw_only=True)
+      class WidgetOperations:
+        id: str
+        weight: int
+
+        @staticmethod
+        def do_work() -> None:
+          """
+          This is a test
+          """
+          pass
+
+
+      `);
+  });
+
+  it("creates a class with abstract method if an interface is provided", async () => {
+    const { program, WidgetOperations } = await Tester.compile(t.code`
+    interface ${t.interface("WidgetOperations")} {
+      op getName(id: string): string;
+    }
+    `);
+
+    expect(getOutput(program, [<ClassDeclaration type={WidgetOperations} />])).toRenderTo(`
+      from abc import ABC
+      from abc import abstractmethod
+      from dataclasses import dataclass
+
+      @dataclass(kw_only=True)
+      class WidgetOperations(ABC):
```

Comment:
It feels a little strange to combine a dataclass with an ABC - as I guess dataclasses are generally associated with being a container for data, rather than containing an interface that needs to be implemented.
What scenario would result in this being generated and how do we anticipate people using it?